ldms/src/core/ldms.c

```diff
@@ -309,9 +309,10 @@ struct get_set_names_arg {
  * }
  */
 
-static const char *perm_string(uint32_t perm)
+#define PERM_BUF(x) char x[16]
```

Comment:
@nichamon It is a common misconception that you can define and have the compiler check string sizes with the syntax you propose. The compiler will happily compile { char x[2]; uint32_t p = 0; perm_string(p, x); }, even in pedantic mode and leave you to enjoy your buffer overrun.

C does, however provide a solution for fixed array sizes: { typedef char(*b_t)[14]; int foo(b_t buf) {...} }. This solution has some annoying side effects (casting where buf is used inside foo), but it does let the compiler trap (warn) about the undersized  buffer in  { char x[2]; uint32_t p = 0; perm_string(p, x); }.  I prefer the macro version, but I will switch to the typedef since it enables compiler checking instead of reading comments or inferring from the function signature.

Reply:
@nichamon I removed the macro-concealed type declaration.